ci/build_docs.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+rapids-logger "Create test conda environment"
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key docs \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
+
+rapids-dependency-file-generator \
+  --output requirements \
+  --file_key docs \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee requirements.txt
+  
+rapids-mamba-retry env create --force -f env.yaml -n docs
+conda activate docs
+pip install -r requirements.txt
+
+rapids-print-env
+
+rapids-logger "Downloading artifacts from previous jobs"
+CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
+PYTHON_CHANNEL=$(rapids-download-conda-from-s3 python)
+VERSION_NUMBER=$(rapids-get-rapids-version-from-git)
+
+rapids-mamba-retry install \
+  --channel "${CPP_CHANNEL}" \
+  --channel "${PYTHON_CHANNEL}" \
+  libraft-distance \
+  libraft-headers \
+  libraft-nn \
+  pylibraft \
+  raft-dask
+
+
+# Build CPP docs
+rapids-logger "Build CPP docs"
+pushd docs
+sphinx-build -b dirhtml source build -W
```

Comment:
RAFT uses a plugin called `breathe` for sphinx, which means cmake also gets involved in the docs build process and the python docs build is combined w/ the c++ docs build. Is there any reason we can't use `./build.sh docs` here?